portal/models/questionnaire_response.py

```diff
@@ -230,19 +231,26 @@ def validate_document(document):
 
     @property
     def document_answered(self):
+        """ Return a modified copy of self.document including answer text
+
+        QuestionnaireResponse populated with text answers based on codes
+        in valueCoding.  This always returns a *copy* of the document so
+        any local or subsequent mutations aren't persisted or found in db
+        session cached objects.
+
         """
-        Return a QuestionnaireResponse populated with text answers based on codes in valueCoding
-        """
-        instrument_id = self.document['questionnaire']['reference'].split('/')[-1]
+        instrument_id = self.document['questionnaire']['reference'].split(
+            '/')[-1]
         questionnaire = Questionnaire.find_by_name(name=instrument_id)
+        document = copy.deepcopy(self.document)
```

Comment:
Not sure how much of a change it would be, but could be more performant if the `deepcopy` happened after the check for a Questionnaire, since we won't need to modify it anyway.

Reply:
not true - look at how it's used in the view function below - the return value gets modified regardless of the questionnaire.